test/test_helper.rb

```diff
@@ -13,6 +13,11 @@ class Minitest::Test
 
   RecordStore.config_path = DUMMY_CONFIG_PATH
 
+  def teardown
+    Provider::DynECT.instance_variable_set(:@dns, nil)
+    Provider::DNSimple.instance_variable_set(:@dns, nil)
```

Comment:
I don't think `clear_session` is needed since we want to cache the connections during normal operation, but in tests we want to always start with a clean slate.